typescript/cli/src/deploy/core.ts

```diff
@@ -350,6 +399,24 @@ async function executeDeploy({
   logBlue(`Agent configs are in ${agentFilePath}`);
 }
 
+/**
+ * Retrieves artifacts file metadata for the current command.
+ * @param dryRun whether or not the current command is being dry-run
+ * @returns the artifacts files
+ */
+function getArtifactsFiles(dryRun: boolean): Array<ArtifactsFile> {
+  const coreDeploymentFile = {
+    filename: dryRun ? 'dry-run_core-deployment' : 'core-deployment',
+    description: 'Contract addresses',
+  };
+  const agentConfigFile = {
+    filename: dryRun ? 'dry-run_agent-config' : 'agent-config',
+    description: 'Agent configs',
+  };
+
+  return [coreDeploymentFile, agentConfigFile];
```

Comment:
for dry runs im not sure how useful the artifacts are?

Reply:
This allows a developer to have evidence of past runs to see any diff in their deployed changes. Also, we may eventually want to use `anvil_dumpState` and `anvil_loadState` to allow developers to reload actual anvil runs to use the same addresses for dry-runs. See [here](https://book.getfoundry.sh/reference/anvil/#custom-methods) for those custom methods